lib/kamal/secrets/adapters/aws_secretsmanager.rb

```diff
@@ -0,0 +1,25 @@
+class Kamal::Secrets::Adapters::AwsSecretsmanager < Kamal::Secrets::Adapters::Base
+  private
+    def login(_account)
+      nil
```

Comment:
There's no sensible default way to implement this is there? Would be nice if we could prompt for MFA from here if that was enabled, but we'd need to know all about the IAM setup for that to work I think.

Reply:
I think there's just too many different ways to authenticate with AWS. I believe the preferred method is SSO using the `aws sso login` command, but the user could also be authenticated with access keys, environment variables, automatically (if they're on an EC2 instance with an IAM Role), or even an external process.

One option would be to check if the user is authenticated with `aws sts get-caller-identity`, and if that command returns an error, attempt to run `aws sso login`. However, this could be confusing if they normally authenticate a different way, or even worse have aws cli v1 installed which doesn't include sso login.

What do you think?

Reply:
Yes let's leave it for now. We could I guess add a separate `SsoAwsSecretsManager` or something like that later.